src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs

```diff
@@ -27,7 +24,7 @@ internal abstract partial class AbstractEditorInlineRenameService
         /// <summary>
         /// Represents information about the ability to rename a particular location.
         /// </summary>
-        private partial class SymbolInlineRenameInfo : IInlineRenameInfoWithFileRename
+        internal partial class SymbolInlineRenameInfo : IInlineRenameInfoWithFileRename
```

Comment:
instead of makign this private, we should see if TS or Xaml implements IInlineRenameInfoWithFileRename.  If not, we can add this `DocumentSpans` field to that guy (which is likely C#/VB specific), and then just check for that interface.  This type can stay private then.

Reply:
Looks like that type was added by @ryzngard in https://github.com/dotnet/roslyn/pull/36041.  Talk to him to be sure, but i believe IInlineRenameInfoWithFileRename is entirely a C#/VB specific interface that you could just change without breaking anyone else.

Reply:
While technically true that TS and Xaml don't use `IInlineRenameInfoWithFileRename`, it seems dubious to add `DocumentSpans` there. It would go on `IInlineRenameInfo` if anything. What are these spans for?

Reply:
so the problem with putting it on IInlineRenameInfo is that it would break TypeScript (since htat is not on the ExternalAccess pattern yet). So IInlineRenameInfoWithFileRename is at least a bit of a back channel to get this back to the InlineRenameService.

> What are these spans for?

The language passes back teh original locations teh symbol being renamed is at.  And the inline-rename service determines if it should automatically fail because those locations aren't even in files that can be edited.

Reply:
How does this differ from what's returned with `FindRenameLocationsAsync`? 

If we _have_ to put it on `IInlineRenameInfoWithFileRename` let's at least add a big comment and work item to move correctly once TS is on the right IVT pattern.